### PR TITLE
KAFKA-13444: Fix OAuthCompatibilityTool help and add SSL options

### PR DESCRIPTION
**Pull request by kirktrue**
The `OAuthCompatibilityTool` is missing the SSL configuration options. In addition, the help text is incorrect.

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

File: tools/src/main/java/org/apache/kafka/tools/OAuthCompatibilityTool.java [hunk 1]
```diff
@@ -208,71 +170,217 @@ public static void main(String[] args) {
...
 
-        if (value != null)
-            configs.put(configsKey, value);
-    }
+        private Argument addArgument(String option, String help) {
+            return addArgument(option, help, String.class);
+        }
+
+        private Argument addArgument(String option, String help, Class<?> clazz) {
+            // Change foo.bar into --foo.bar.
```
Comment (junrao) [inline tools/src/main/java/org/apache/kafka/tools/OAuthCompatibilityTool.java:257]: This comment is no longer true.

**Review by junrao — COMMENTED** [review]
@kirktrue : Thanks for the PR. Just one comment below.

File: tools/src/main/java/org/apache/kafka/tools/OAuthCompatibilityTool.java (hunk 1 above)
Comment (kirktrue) [reply tools/src/main/java/org/apache/kafka/tools/OAuthCompatibilityTool.java:257]: `option` is `foo.bar`. `name` is `--foo.bar`. `name` is being passed in as the main argument name. So I think the comment is correct unless I'm misunderstanding your comment.

**Review by junrao — APPROVED** [review]
@kirktrue : Thanks for the explanation. LGTM

Since this changes the command line interface in an incompatible way, I guess you need the PR to be in 3.1.0 too?

**Comment by junrao** [comment]
@kirktrue : Could you let David Jacot know if this needs to be in 3.1.0? Thanks.

**Comment by kirktrue** [comment]
@dajac - can you merge this into the 3.1 branch as well? Thanks.

**Comment by dajac** [comment]
@kirktrue @junrao I am OK with merging this to 3.1 branch as it changes the command line interface in an incompatible way. @junrao Could you cherry-pick it to 3.1?

**Comment by junrao** [comment]
merged the PR to 3.1 branch

**Comment by kirktrue** [comment]
Thanks @junrao!

**Pull request merged**